### PR TITLE
QE: Move proxy registration to init_clients

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This will 
- move the proxy registration and onboarding from `core` to `init_clients`. This way, we properly sync the required channels for the proxy before registering it.
- create a new activation key for the proxy, with the correct channels assigned after they are synced.
- modify the `When(/^I execute mgr\-bootstrap "([^"]*)"$/)` step to be more generic, so that we can provide it the activation key we want , instead of always using the default one.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests were adjusted

- [x] **DONE**

## Links

Manager 4.3:

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by nodeg** [comment]
The one test failing looks to me like a flaky test.
image

File: testsuite/features/step_definitions/command_steps.rb [hunk 1]
```diff
@@ -449,8 +449,7 @@
 end
 
 When(/^I execute mgr\-bootstrap "([^"]*)"$/) do |arg1|
-  arch = 'x86_64'
-  $command_output, _code = $server.run("mgr-bootstrap --activation-keys=1-SUSE-KEY-#{arch} #{arg1}")
+  $command_output, _code = $server.run("mgr-bootstrap #{arg1}")
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:452]: So we don't need to provide the activation key anymore here?

Comment (nodeg) [reply testsuite/features/step_definitions/command_steps.rb:452]: Yes, we do not provide a default key there, anymore. Instead, we can now specify which key we want:

```gherkin
When I execute mgr-bootstrap "--activation-keys=1-PROXY-KEY-x86_64 --script=bootstrap-proxy.sh"
```

File: testsuite/features/init_clients/proxy_register_as_minion_with_script.feature [hunk 2]
```diff
@@ -22,15 +22,15 @@ Feature: Setup Uyuni proxy
...
+    When I execute mgr-bootstrap "--activation-keys=1-PROXY-KEY-x86_64 --script=bootstrap-proxy.sh"
     Then I should get "* bootstrap script (written):"
     And I should get "    '/srv/www/htdocs/pub/bootstrap/bootstrap-proxy.sh'"
     When I fetch "pub/bootstrap/bootstrap-proxy.sh" to "proxy"
     And I run "sh ./bootstrap-proxy.sh" on "proxy"
 
 @salt_bundle
   Scenario: Create the bundle-aware bootstrap script for the proxy and use it
-    When I execute mgr-bootstrap "--script=bootstrap-proxy.sh --force-bundle"
+    When I execute mgr-bootstrap "--activation-keys=1-PROXY-KEY-x86_64 --script=bootstrap-proxy.sh"
```
Comment (srbarrios) [inline testsuite/features/init_clients/proxy_register_as_minion_with_script.feature:33]: We are making this scenario Architecture dependent. For now, no problem, as we only support SUSE Manager Proxy with X86_64, but I bet we will be supporting Proxy with ARM at some point soon.

Comment (nodeg) [reply testsuite/features/init_clients/proxy_register_as_minion_with_script.feature:33]: FMPOV it was architecture dependent before where we had the architecture hard coded in the step. With my changes, you are able to provide the activation key you want regardless of the architecture. The proxy key I created here is simply only for x86_64.

**Review by srbarrios — APPROVED** [review]

**Comment by nodeg** [comment]
> You might want to add the list of synced proxy channels to `constants.rb` , I believe we still kill the sync of unlisted channels IIRC

You mean here? https://github.com/uyuni-project/uyuni/blob/master/testsuite/features/support/constants.rb#L479-L487

**Comment by elariekerboull** [comment]
After looking a bit more into it I'm not sure if that part is useless now since in theory we sync only what we want, I'll check
Otherwise if I'm not mistaken it's in CHANNEL_TO_SYNCH_BY_OS_VERSION

**Comment by nodeg** [comment]
Furthermore, I was wondering what to add. I took a look at SCC and saw
image
image

There I would go for the second one (`SUSE Manager Proxy Module 4.3 x86_64`)

**Comment by jordimassaguerpla** [comment]
Re the github validation, this is fine :+1:

**Comment by nodeg** [comment]
Oscar and I concluded to add the following:

```bash
sle-module-suse-manager-proxy-4.3-pool-x86_64
sle-module-suse-manager-proxy-4.3-updates-x86_64
sle-module-server-applications15-sp4-pool-x86_64-proxy-4.3
sle-module-server-applications15-sp4-updates-x86_64-proxy-4.3
sle-module-basesystem15-sp4-pool-x86_64-proxy-4.3
sle-module-basesystem15-sp4-updates-x86_64-proxy-4.3
sle-product-suse-manager-proxy-4.3-pool-x86_64
sle-product-suse-manager-proxy-4.3-updates-x86_64
```

**Review by maximenoel8 — APPROVED** [review]

**Comment by nodeg** [comment]
The channels were assigned correctly to the activation key and do show up after onboarding the proxy:
image

**Comment by nodeg** [comment]
I will squash everything in 2 commits after the tests are done.

**Review by ktsamis — APPROVED** [review]

**Pull request merged**